torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -1289,12 +1294,22 @@ def test_all_reduce_result_cuda(self):
                 self.assertEqual(result, [_build_tensor(src + 1, expected_value)])
             self._barrier()
 
-        def call_dist_op(self, profiling_title_postfix, is_async, op, *args, expect_event=True, secondary_op_call=None, **kwargs):
+        def call_dist_op(
+            self,
+            profiling_title_postfix,
+            is_async,
+            op,
+            *args,
+            expect_event=True,
+            secondary_op_call=None,
+            profile_cuda=False,
+            **kwargs,
+        ):
             op_calls = [lambda: op(*args, **kwargs)]
             if secondary_op_call is not None:
                 op_calls.append(secondary_op_call)
 
-            with torch.autograd.profiler.profile() as prof:
+            with torch.autograd.profiler.profile(use_cuda=profile_cuda) as prof:
```

Comment:
qq, does `use_cuda` change any info on the event? Is there a way to verify that `use_cuda` does make a difference here?

Reply:
It doesn't change any info on the `gloo::allreduce` or other collectives since they are profiled with `RecordFunction`, although the aten ops such as `copy` are appropriately recorded with cuda time. I think asserting on that is better covered by profiler specific tests though, here we just want to ensure that enabling cuda profiling does not cause any issues.